Project/src/pages/SpeakingMode/SpeakingMode.ts

```diff
@@ -0,0 +1,189 @@
+import { Component, NgZone } from '@angular/core';
+import { NavController, NavParams, Platform, ToastController } from 'ionic-angular';
+import { Util } from '../../util';
+import { MediaPlugin } from 'ionic-native';
+import { screenUnit } from '../../interfaces';
+
+
+declare var cordova: any;
+declare var SpeechRecognition: any;
+
+@Component({
+    selector: 'page-SpeakingMode',
+    templateUrl: 'SpeakingMode.html'
+})
+export class SpeakingMode {
+    recognition: any;
+    platform: any;
+    title: string; // Title of the session
+    isCorrect: any;
+    public currUnit: screenUnit; // Current screenUnit
```

Comment:
Don't think these need to be "public." They're not public in the listening mode file.